src/vs/workbench/contrib/scm/browser/scmViewPane.ts

```diff
@@ -1182,6 +1182,20 @@ class ViewModel {
 			if (e.reason === WillSaveStateReason.SHUTDOWN) {
 				this.storageService.store(`scm.viewState`, JSON.stringify(this.treeViewState), StorageScope.WORKSPACE, StorageTarget.MACHINE);
 			}
+
+			this.mode = this.getViewModelMode();
```

Comment:
Why do we get the mode and sort key in `onWillSaveState`?

Reply:
`this.mode` and `this.sortKey` are setters which will write the latest values of these keys to the storage service.

Reply:
Oh wow 🤯 . I did not write this code, that explains. I think `set` should never have side effects like that.